lib/registrant.rb

```diff
@@ -0,0 +1,43 @@
+class Registrant
+  
+  attr_reader :name,
+              :age,
+              :permit,
+              :license_data
+
+  def initialize(name, age, permit = false)
+    @name = name
+    @age = age
+    @permit = permit
+    @license_data = {
+      written: false,
+      license: false,
+      renewed: false
+    }
+  end
+
+  def permit?
+    @permit
+  end
+
+  def license_data?
+    @license_data
+  end
+
+  def earn_permit
+    @permit = true
+  end
+
+  def written_test_complete
```

Comment:
I really love this and the following methods that set the value of license_data keys. This is a great example of encapsulation. We *could* use an `attr_accessor` for the license_data attribute and let other classes set the values for license_data (and if you did this it would be fine), but we often talk about how attr_accessors are not always the best choice because they give the code outside of this class a lot of control to set the values. What if they set `license_data[:banana] = true`? What's stopping that from happening! Instead, you're keeping license_data to read-only permission, and controlling the value of these hash keys with these methods, which we call `setter` methods (because they set the value).